src/modules/configs/config.py

```diff
@@ -75,3 +75,119 @@ class ApplicationConf(Conf):
 
     def __init__(self, conf_path: Union[str, bool]):
         super().__init__(conf_path, "application")
+
+
+# pylint: disable=too-many-instance-attributes
+class MergedConf:
+    """Class for merging configuration files"""
+
+    def __init__(self, logger: Logger, options: dict):
+        self.logger = logger
+        self.options = options
+
+        self.operator = OperatorConf(options["operator_conf"]).parse_conf()
+        self.project = ProjectConf(options["project_conf"]).parse_conf()
+        self.application = ApplicationConf(options["application_conf"]).parse_conf()
+
+        self.operator_keys = self.flatten_dict(
+            self.operator, exclude_keys=global_vars.config_rules["exclude_flatten"])
+        self.project_keys = self.flatten_dict(
+            self.project, exclude_keys=global_vars.config_rules["exclude_flatten"])
+        self.application_keys = self.flatten_dict(
+            self.application, exclude_keys=global_vars.config_rules["exclude_flatten"])
+
+        self.final = self.create_final_conf()
+
+    def flatten_dict(self, dct, parent_key='', sep='_', exclude_keys=None):
+        if exclude_keys is None:
+            exclude_keys = []
+        flattened = {}
+        for key, value in dct.items():
+            if key in exclude_keys:
+                flattened[key] = value
+            else:
+                new_key = f"{parent_key}{sep}{key}" if parent_key else key
+                if isinstance(value, list):
+                    for idx, item in enumerate(value):
+                        item_key = f"{new_key}{sep}{idx}"
+                        if isinstance(item, dict):
+                            flattened.update(self.flatten_dict(item, item_key, sep))
+                        else:
+                            flattened[item_key] = item
+                elif isinstance(value, dict):
+                    flattened.update(self.flatten_dict(value, new_key, sep))
+                else:
+                    flattened[new_key] = value
+        return flattened
+
+    def get_all_conf(self):
```

Comment:
It's used in 1 place only, so I think we can remove and call directly below.

Reply:
Also `for key in dict:` is supposedly faster than `for x in dict.keys()`. Details in this PEP if you're interested. https://peps.python.org/pep-0234
For sure I haven't read it but I trust the 2nd answer here. ;) https://stackoverflow.com/questions/3294889/iterating-over-dictionaries-using-for-loops.

Reply:
Also, `templates.final` is the template for the config that we want returned form this module. After merging confs, values from `merge_conf()` need to be mapped to `templates.final`.

Reply:
So we need a test function for merge_conf() to test if the result is mapped to templates.final?